v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -233,11 +228,20 @@ function getTaskFromCliArguments(
     const arg = cliArguments[i];
 
     if (arg.startsWith("--")) {
+      // A standalone '--' is ok because it is used to separate CLI tool arguments from script arguments,
+      // ensuring the tool passes subsequent options directly to the script.
       if (arg.length === 2 || task !== undefined) {
         break;
       }
 
-      throw new Error(`Found task argument ${arg} before the task name`);
+      // At this point in the code, the global parameters have already been parsed, so the remaining parameters starting with '--' are task named parameters.
+      // Hence, if no task is defined, it means that the parameter is not assigned to any task, and it's an error.
+      throw new HardhatError(
+        HardhatError.ERRORS.ARGUMENTS.UNRECOGNIZED_NAMED_PARAM,
```

Comment:
I think this error is too general. In this case we are almost sure that the user accidentally added a non-global param before the task name, so it's better to throw a more specialized error.

Reply:
Great point!

Reply:
So should we keep the current error message?